src/Compilers/Test/Utilities/CSharp/NullableAttributesVisitor.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp.Test.Utilities
+{
+    /// <summary>
+    /// Returns a string with all symbols containing nullable attributes.
+    /// </summary>
+    internal sealed class NullableAttributesVisitor : CSharpSymbolVisitor
+    {
+        internal static string GetString(Symbol symbol)
+        {
+            var builder = new StringBuilder();
+            var visitor = new NullableAttributesVisitor(builder);
+            visitor.Visit(symbol);
+            return builder.ToString();
+        }
+
+        private readonly StringBuilder _builder;
+        private readonly HashSet<Symbol> _reported;
+
+        private NullableAttributesVisitor(StringBuilder builder)
+        {
+            _builder = builder;
+            _reported = new HashSet<Symbol>();
+        }
+
+        public override void DefaultVisit(Symbol symbol)
+        {
+            ReportSymbol(symbol);
+        }
+
+        public override void VisitModule(ModuleSymbol module)
+        {
+            Visit(module.GlobalNamespace);
+        }
+
+        public override void VisitNamespace(NamespaceSymbol @namespace)
+        {
+            VisitList(@namespace.GetMembers());
+        }
+
+        public override void VisitNamedType(NamedTypeSymbol type)
+        {
+            ReportSymbol(type);
+            VisitList(type.TypeParameters);
+            VisitList(type.GetMembers());
+        }
+
+        public override void VisitMethod(MethodSymbol method)
+        {
+            // Skip accessors since those are covered by associated symbol.
+            if (method.IsAccessor()) return;
+
+            ReportSymbol(method);
+            VisitList(method.TypeParameters);
+            VisitList(method.Parameters);
+        }
+
+        public override void VisitTypeParameter(TypeParameterSymbol typeParameter)
+        {
+            ReportSymbol(typeParameter);
+        }
+
+        private void VisitList<TSymbol>(ImmutableArray<TSymbol> symbols) where TSymbol : Symbol
+        {
+            foreach (var symbol in symbols)
+            {
+                Visit(symbol);
+            }
+        }
+
+        private static string GetIndentString(Symbol symbol)
+        {
+            int level = 0;
+            while (true)
+            {
+                symbol = symbol.ContainingSymbol;
+                if (symbol.Kind == SymbolKind.Namespace)
+                {
+                    break;
+                }
+                level++;
+            }
+            return new string(' ', level * 4);
+        }
+
+        private static readonly SymbolDisplayFormat _displayFormat =
```

Comment:
>_displayFormat [](start = 52, length = 14)

Assuming this is a variant of an existing format, could it be expressed as `ExistingFormat.With(change)` instead? I think that is easier to maintain when we add new options.